modules/drouting/drouting.c

```diff
@@ -1266,36 +1269,38 @@ static inline int dr_reload_data_head(struct head_db *hd,
 
 	/* destroy old data */
 	if (old_data) {
-		/* copy the state of gw/cr from old data */
-		/* interate new gws and search them into old data */
-		for (map_first(new_data->pgw_tree, &it);
-				iterator_is_valid(&it); iterator_next(&it)) {
-			dest = iterator_val(&it);
-			if(dest==NULL)
-				break;
+		if (is_inherit_state) {
+			/* copy the state of gw/cr from old data */
+			/* interate new gws and search them into old data */
+			for (map_first(new_data->pgw_tree, &it);
+					iterator_is_valid(&it); iterator_next(&it)) {
+				dest = iterator_val(&it);
+				if(dest==NULL)
+					break;
 
-			gw=(pgw_t *)*dest;
+				gw=(pgw_t *)*dest;
```

Comment:
please check if the indentation is with spaces  and if so, replace with TAB  `\t`

Reply:
i checked, this is tab